File: frontend/src/app/services/veicolo.service.ts
import { Injectable } from '@angular/core';
import { Veicolo } from '../models/veicolo.model';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class VeicoloService {

  private baseUrl = '/Veicolo';

  constructor(private http: HttpClient) {}

  getAll(): Observable<Veicolo[]> {
    return this.http.get<Veicolo[]>(`${this.baseUrl}/getVeicoli`);
  }


  getById(id: number): Observable<Veicolo> {
    return this.http.get<Veicolo>(`${this.baseUrl}/getVeicoloById/${id}`);
  }


  create(v: Veicolo): Observable<Veicolo> {
    return this.http.post<Veicolo>(`${this.baseUrl}/insVeicolo`, v);
  }


  update(v: Veicolo): Observable<Veicolo> {
    return this.http.put<Veicolo>(`${this.baseUrl}/updVeicolo`, v);
  }


  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/deleteVeicoloById/${id}`);
  }

}

File: backend/flexikmservlet/src/main/java/alexpre/flexikm/servlet/service/VeicoloService.java
package alexpre.flexikm.servlet.service;

import java.util.List;

import org.springframework.stereotype.Service;

import alexpre.flexikm.servlet.model.Veicolo;

@Service
public interface VeicoloService
{
  public Veicolo insert(Veicolo veicolo);

  public List<Veicolo> getVeicoli();

  public Veicolo update(Veicolo garanzia);

  public String deleteVeicoloById(Long id);

  public List<Veicolo> getVeicoloById(Long id);

}
